index.tt

```diff
@@ -6,152 +6,349 @@
+    <!-- <a class="btn btn-large btn-success" href="[% root %]download.html">Find Packages</a> -->
+    <!-- <a class="btn btn-large btn-success" href="https://github.com">View on GitHub</a> -->
+  </div>
+
+  <div class="span6">
+    <h4>TODO: A video / picture of NixOS in action!</h4>
+    <!-- 
+
+    -->
+  </div>
+</div>
+
+<hr />
+
+<ul id="features">
+  <li class="active"><a href="#develop">Develop</a></li>
+  <li><a href="#build">Build</a></li>
+  <li><a href="#deploy">Deploy</a></li>
+</ul>
+
+<div id="features-content" class="tab-content">
+  <section class="tab-pane active" id="develop">
+    <div class="row-fluid">
+      <div class="span6">
+
+        <h3>Try new tools without fear</h3>
+        <p>Don't clutter your system with tools that you use only now and then.</p>
+        <pre class="well" style="margin-top:1em">
+<span class="nix-command">python --version</span>
+python: command not found
+<span class="nix-command"><strong>nix-shell -p python3</strong></span>
+(nix-shell)<span class="nix-command">python --version</span>
+Python 3.7.7</pre>
+
+        <h3>One tool, multiple languages</h3>
+        <p></p>
+        <pre class="well" style="margin-top:1em">
+<span class="nix-command"><strong>nix-shell -p python3 nodejs go rustc</strong></span>
+(nix-shell)<span class="nix-command">node --version</span>
+v10.20.1
+(nix-shell)<span class="nix-command">go version</span>
+go version go1.14.1 linux/amd64
+(nix-shell)<span class="nix-command">rustc --version</span>
+rustc 1.42.0</pre>
+
       </div>
-    </div>
-    <div class="intro-nixos">
-      <h2>What is NixOS?</h2>
-    
-      <p>NixOS is a Linux distribution with a unique approach
-      to package and configuration management. Built on top of the
-      Nix package manager, it is completely declarative, makes
-      upgrading systems reliable, and has
-      <a href="[%root%]nixos/about.html">many other advantages</a>.</p>
-    
-      <div class="learn-more">
-        <a class="btn btn-success" href="[%root%]nixos/">
-          <i class="fa fa-chevron-right"></i> Learn more about NixOS
-        </a>
+      <div class="span6">
+
+        <h3>Isolated development environments</h3>
+        <p>
+          After you get familiar with <code>nix-shell -p</code> you can take
+          the next step further and <a href="[%root%]learn.html">learn some
+            Nix</a>. To setup a more persistent environment you can also write a
+          simple <code>shell.niix</code> file:
+        </p>
+        <pre class="well" style="margin-top:1em">
+{ pkgs ? import &lt;nixpkgs&gt; {}
+}:
+pkgs.mkShell {
+  name = "dev-shell";
+  buildInputs = [
+    pkgs.python3
+    pkgs.python3Packages.virtualenv
+    pkgs.nodejs
+    pkgs.yarn
+  ];
+}</pre>
+        <p>Then enter development environment with:</p>
+        <pre class="well" style="margin-top:1em">
+<span class="nix-command"><strong>nix-shell</strong></span>
+(nix-shell)<span class="nix-command">virtualenv --version</span>
```

Comment:
before publishing this we should check if the tools like virtualenv actually work as expected